src/Servers/Kestrel/Transport.Quic/src/Internal/QuicStreamContext.cs

```diff
@@ -371,7 +379,7 @@ private async Task DoSend()
         // A client can abort a stream after it has finished sending data. We need a way to get that notification
         // which is why we listen for a notification that the write-side of the stream is done.
         // An exception can be thrown from the stream on client abort which will be captured and then wake up the output read.
-        _ = WaitForWritesCompleted();
+        var waitForWritesClosedTask = WaitForWritesClosedAsync();
```

Comment:
Bad merge? This undoes https://github.com/dotnet/aspnetcore/pull/42683

Reply:
Yeah, I noticed some failures after that change. Also, awaiting the task is required to pool it.

Reply:
`_ =` for Tasks and ValueTasks is almost always bad. I've seen very few counterexamples. Although it's often done for expedience when properly handling the task would be extremely difficult, it would still be better to await everything if it were easily doable.

Even if you do things like log all uncaught exceptions to make discarding tasks better, how do you know the logger is still alive sending your logs to wherever they need to go if you aren't awaiting all your tasks and properly delaying shutdown?